packages/ui/src/theme/components/utils.ts

```diff
@@ -35,12 +35,8 @@ export type Modifiers<
   Keys extends string = string,
   Required extends boolean = false,
 > = Required extends true
-  ? {
-      _modifiers: { [key in Keys]: ComponentStyles };
-    }
-  : {
-      _modifiers?: { [key in Keys]?: ComponentStyles };
-    };
+  ? { _modifiers: Record<Keys, ComponentStyles> }
+  : { _modifiers?: Partial<Record<Keys, ComponentStyles>> };
```

Comment:
Addresses the [same linting issue as these types have](https://github.com/aws-amplify/amplify-ui/pull/6521/commits/1f2d66ce86fe5a675ef8ec6a76a574966d5e84d9#r2052985817), but was able to actually resolve it since `Keys` is just being used to key the `_modifiers` and not for iterating over